docs/guides/auth.md

```diff
@@ -175,26 +175,41 @@ var scope = 'https://www.googleapis.com/auth/grpc-testing';
 
 ```csharp
 // Base case - No encryption/authentication
```

Comment:
This looks really good - my only suggestion (given that there are now multiple possible approaches for auth with Google) is that for clarity maybe we should split this into multiple code blocks (no auth, recommended approach with JWT access token, legacy approach with OAuth2, and per-RPC) with headings rather than just comments within a big code block.

It's also not super clear how each approach ties in with the generic "Authenticating with Google" paragraph up yonder, though that might be because it also needs some work - who should I prod about doing a separate PR for that?